core/src/mindustry/world/consumers/ConsumeItems.java

```diff
@@ -39,7 +39,10 @@ public void build(Building tile, Table table){
             for(ItemStack stack : items){
                 c.add(new ReqImage(new ItemImage(stack.item.icon(Cicon.medium), stack.amount),
                 () -> tile.items != null && tile.items.has(stack.item, stack.amount))).padRight(8);
-                if(++i % 4 == 0) table.row();
+
+                if(i++ % 5 == 4){
```

Comment:
Why is the math here different?

Reply:
the previous math did not create new rows

Reply:
There's nothing wrong with the math (`++i % 4 == 0`), only the table needs to be changed.

Reply:
If there's nothing wrong with the math, why doesn't it create new rows?
![image](https://user-images.githubusercontent.com/54301439/116797627-789b5d00-aa9c-11eb-9064-c921201ac2b4.png)

I checked the v1.26.2 tag, it has that math on it.

It could be because the `++` was before the variable?

Whatever the case, the equation is like that of the one in ItemSelection.

Reply:
> If there's nothing wrong with the math, why doesn't it create new rows?

Because there's a typo, it creates rows for the **parent** table, not the table that actually has the icons in it.